src/map.rs

```diff
@@ -250,7 +240,10 @@ mod tests {
         assert_eq!(server.socket_file(), sock_file);
 
         let (shutdown_tx, shutdown_rx) = oneshot::channel();
-        let task = tokio::spawn(async move { server.start_with_shutdown(shutdown_rx).await });
+        let shutdown = async {
+            shutdown_rx.await.unwrap();
+        };
+        let task = tokio::spawn(async move { server.start_with_shutdown(shutdown).await });
 
         tokio::time::sleep(Duration::from_millis(50)).await;
```

Comment:
i forget why we need this sleep.

Reply:
This was done just to ensure the server is started completely before sending requests to it.